spyder/plugins/editor/widgets/editor.py

```diff
@@ -2516,6 +2541,117 @@ def dropEvent(self, event):
         event.acceptProposedAction()
 
 
+class AutosaveComponent:

```

Comment:
* Please also move this class to `plugins/editor/utils/autosave.py`.
* Also, please make this inherit from `object` to make it a new class in Python 2.